.github/workflows/test.yml

```diff
@@ -12,13 +12,41 @@ env:
 jobs:
   library:
     name: Library
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
     steps:
       - name: Checkout project sources
         uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '11'
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
       - name: Validate Gradle Wrapper
         uses: gradle/wrapper-validation-action@v1
-      - name: Gradle Run Tests
+      - name: Install docker
+        run: |
+          mkdir -p ~/.docker/machine/cache
+          curl -Lo ~/.docker/machine/cache/boot2docker.iso https://github.com/boot2docker/boot2docker/releases/download/v19.03.12/boot2docker.iso
+          brew install docker docker-compose
+      - name: Start colima
+        run: |
+          # Docker engine is no longer available because of licensing
+          # Alternative Colima is part of the github macOS runner
+          colima start
+          # Ensure colima is configured for later user
+          echo "DOCKER_HOST=unix://$HOME/.colima/default/docker.sock" >> $GITHUB_ENV
+          # Verify Docker
+          docker ps
+      - name: Start local test server
+        run: docker-compose -p xmtp -f dev/local/docker-compose.yml up -d
+      - name: Gradle Run Unit Tests
         run: ./gradlew library:testDebug
+      - name: Gradle Run Integration Tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 29
+          script: ./gradlew connectedCheck
```

Comment:
The more I think about this the more I don't think we need this just yet. The actual full integration test on the device will come once I have an example app but for now my integration tests are just running on the local node which doesn't necessarily need the emulator.